doc/api/single-executable-applications.md

```diff
@@ -0,0 +1,44 @@
+# Single executable applications
```

Comment:
In addition to this file I think we need something like [doc/contributing/maintaining-single-executable-application-support.md](https://github.com/nodejs/node/pull/42334/files#diff-52afcfcc40fdf2bc1f780ed830626b5a53d652721c04324140d62c87e50087c0)  which explains the strategy, layout of what's injected etc.